### PR TITLE
Added FP32 to FP16 conversion transformation

### PR DESCRIPTION
**Pull request by GlebKazantaev**

### CODE REVIEW

File: src/ngraph/pass/convert_fp32_to_fp16.cpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
+//*****************************************************************************
+// Copyright 2020 Intel Corporation
```
Comment (rkimballn1) [inline src/ngraph/pass/convert_fp32_to_fp16.cpp]:
```suggestion
// Copyright 2017-2020 Intel Corporation
```

File: src/ngraph/pass/convert_fp32_to_fp16.hpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
+//*****************************************************************************
+// Copyright 2020 Intel Corporation
```
Comment (rkimballn1) [inline src/ngraph/pass/convert_fp32_to_fp16.hpp]:
```suggestion
// Copyright 2017-2020 Intel Corporation
```

File: src/ngraph/pass/convert_fp32_to_fp16.hpp [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    {
+        convert_constants_precision();
+        convert_parameters_precision();
+    }
+
+private:
+    void convert_constants_precision();
+
+    void convert_parameters_precision();
+};
```
Comment (rkimballn1) [inline src/ngraph/pass/convert_fp32_to_fp16.hpp]:
```suggestion
};

```

**Comment by GlebKazantaev** [comment]
@diyessi can you suggest what is wrong in CI ? Just can't find any error message.

**Pull request merged**